qiime2/sdk/tests/test_bad_inputs.py

```diff
@@ -0,0 +1,161 @@
+# ----------------------------------------------------------------------------
+# Copyright (c) 2016-2019, QIIME 2 development team.
+#
+# Distributed under the terms of the Modified BSD License.
+#
+# The full license is in the file LICENSE, distributed with this software.
+# ----------------------------------------------------------------------------
+
+import os
+import collections
+import tempfile
+import qiime2.core.archive as archive
+
+from qiime2.core.testing.util import get_dummy_plugin
+from qiime2.plugin.testing import TestPluginBase
+
+from qiime2.sdk import Artifact, Visualization
+from qiime2.core.testing.type import IntSequence1, IntSequence2, SingleInt
+from qiime2.core.testing.visualizer import most_common_viz
+from qiime2 import Metadata
+from qiime2.metadata.tests.test_io import get_data_path
+
+
+class TestBadInputs(TestPluginBase):
```

Comment:
I'm not sure this test case needs to be in a new test file (`test_bad_inputs.py`) --- maybe it could get moved into `test_method.py`?

Reply:
These tests all look good to me, I _think_ you have all the cases covered! I will take a closer look in round 2 of review!

Reply:
> I'm not sure this test case needs to be in a new test file (`test_bad_inputs.py`) --- maybe it could get moved into `test_method.py`?

These tests use a couple of visualizers and pipelines with features not available in our current dummy-plugin methods. Am I right to assume that these non-method tests shouldn't live in `test_method.py'? I kinda like breaking these out thematically, but understand the organizational slippery slope there. 

Is it worth adding/modifying a method with `choices` and `range` predicates to fill the gap, then bringing these into `test_method`?

Reply:
How about you split these new tests up? Methods into `test_method.py`, Visualizers into `test_visualizer.py`, Pipelines into `test_pipeline.py`. These tests could still be "grouped" in a `TestBadInputs` suite in each respective file.

Reply:
Renamed file to `test_action.py` per our discussion, to align with sdk test naming conventions and remove the need to break up test functions that use more than one type of Action to test type mismatches.